iup/eval.py

```diff
@@ -1,117 +1,96 @@
-from iup import IncidentUptakeData, PointForecast
 import polars as pl
+from iup import IncidentUptakeData, PointForecast
+from typing import Callable
 
 
-def get_mspe(data: IncidentUptakeData, pred: PointForecast) -> pl.DataFrame:
+###### evaluation metrics #####
+def check_date_match(data: IncidentUptakeData, pred: PointForecast):
     """
-    Calculate MSPE between data and pred
+    Check the dates between data and pred.
+    Dates must be 1-on-1 equal and no duplicate.
     ----------------------
-    Input: data, pred in a matching and validated format
-    Return: pl.DataFrame with MSPE and the forecast start date and end date
+
+    Parameters
+    data:
+        The observed data used for modeling. Should be IncidentUptakeData
+    pred:
+        The forecast made by model. Should be PointForecast
+
+    Return
+    Error if conditions fail to meet.
 
     """
-    # Check the conditions for date match:
-    # 1. Mutual dates must exist between data and prediction.
-    assert any(
-        data["date"].is_in(pred["date"])
-    ), "No matched dates between data and prediction."
+    # sort data and pred by date #
+    data = data.sort("date")
+    pred = pred.sort("date")
 
-    # 2. There should not be any duplicated date in either data or prediction.
-    common_dates = data.filter(pl.col("date").is_in(pred["date"])).select("date")
+    # 1. Dates must be 1-on-1 equal
+    (data["date"] == pred["date"]).all()
 
+    # 2. There should not be any duplicated date in either data or prediction.
     assert (
-        len(common_dates) == common_dates.n_unique()
-    ), "Duplicated dates are found in data or prediction."
+        len(data["date"]) == data["date"].n_unique()
```

Comment:
[low priority] Another way to do this is with https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.is_duplicated.html 

I think it would be:
```python
assert data["date"].is_duplicated().any().not_()
```